### PR TITLE
Allow py-py2cairo to be built with cairo~X

### PR DESCRIPTION
**Pull request by healther**

### CODE REVIEW

**Comment by citibeth** [comment]
I agree with @adamjstewart.  Forwarding variants buys us little functionality, and the space of possible variants one COULD forward in any given package is practically infinite.  Best to not forward variants, and rely on Spack to get what we need.

File: var/spack/repos/builtin/packages/py-py2cairo/package.py [hunk 1]
```diff
@@ -33,9 +33,13 @@ class PyPy2cairo(Package):
 
     version('1.10.0', '20337132c4ab06c1146ad384d55372c5')
 
+    variant('X', default=True,
+            description='Enable cairo X11 dependency')
+
     extends('python')
 
-    depends_on('cairo+X')
```
Comment (citibeth) [inline var/spack/repos/builtin/packages/py-py2cairo/package.py]: Please change to `depends_on('cairo')`

File: var/spack/repos/builtin/packages/py-py2cairo/package.py [hunk 2]
```diff
@@ -33,9 +33,13 @@ class PyPy2cairo(Package):
 
     version('1.10.0', '20337132c4ab06c1146ad384d55372c5')
 
+    variant('X', default=True,
+            description='Enable cairo X11 dependency')
+
```
Comment (citibeth) [inline var/spack/repos/builtin/packages/py-py2cairo/package.py]: Please remove this variant.
[Also... variants need to be lower-case name; although we haven't gotten around to fixing past transgressions yet.]

**Comment by adamjstewart** [comment]
Actually, now that I think about it, I was the one who added the `cairo+X` dependency to this package in #2778. If I remember correctly, `py2cairo ^cairo~X` (the default) didn't build for me at all and complained that cairo _had_ to be built with X11 support. You should confirm that you can actually build `py2cairo ^cairo~X`, preferably on a machine without X11 support builtin.

File: var/spack/repos/builtin/packages/py-py2cairo/package.py (hunk 2 above)
Comment (healther) [inline var/spack/repos/builtin/packages/py-py2cairo/package.py]: why should the variant be removed? Isn't that the usecase? One variant uses the X11 backend the other doesn't?

**Comment by healther** [comment]
@adamjstewart both ```py-py2cairo ^cairo+X``` and ```py-py2cairo ^cairo~X``` build for me

**Comment by citibeth** [comment]
>
>
> why should the variant be removed? Isn't that the usecase? One variant
> uses the X11 backend the other doesn't?
>

Because the variant is not used by the package itself, it's just
forwarded.  The use case is already covered without the variant.

**Comment by adamjstewart** [comment]
@healther I understand your point, but how far do we go? Should we forward all variants up the DAG for every package? If A depends on B depends on C, and C has a `+shared` variant, should A and B also have `+shared` variants that tell C how it should be built? Obviously not, and Spack provides a simple way of specifying how you want C built:
```
$ spack spec A ^C+shared
```
You can also put this in your `~/.spack/packages.yaml` to make it permanent.

**Comment by adamjstewart** [comment]
I'm trying to build `py-py2cairo ^cairo~X` now on my cluster which doesn't have any X11 libs installed on the system.

**Pull request merged**